mail_tracking/models/mail_message.py

```diff
@@ -110,16 +124,82 @@ def tracking_status(self):
             res[message.id] = partner_trackings
         return res
 
+    @api.multi
+    def _get_failed_message(self):
+        res = {}
+        for message in self:
+            res.update({
+                message.id: message.mail_tracking_needs_action
+                and bool(message.mail_tracking_ids.filtered(
+                    lambda x: x.state in self.get_failed_states()))
+            })
+        return res
+
     @api.model
     def _message_read_dict_postprocess(self, messages, message_tree):
         res = super(MailMessage, self)._message_read_dict_postprocess(
             messages, message_tree)
         mail_message_ids = {m.get('id') for m in messages if m.get('id')}
         mail_messages = self.browse(mail_message_ids)
         partner_trackings = mail_messages.tracking_status()
+        failed_message = mail_messages._get_failed_message()
         for message_dict in messages:
             mail_message_id = message_dict.get('id', False)
             if mail_message_id:
-                message_dict['partner_trackings'] = \
-                    partner_trackings[mail_message_id]
+                message_dict.update({
+                    'partner_trackings': partner_trackings[mail_message_id],
+                    'failed_message': failed_message[mail_message_id],
+                })
         return res
+
+    @api.model
+    def _prepare_dict_failed_message(self, message):
+        failed_trackings = message.mail_tracking_ids.filtered(
+            lambda x: x.state in self.get_failed_states())
+        failed_partners = failed_trackings.mapped('partner_id')
+        failed_recipients = failed_partners.name_get()
+        return {
+            'id': message.id,
+            'date': message.date,
+            'author_id': message.author_id.name_get()[0],
```

Comment:
```suggestion
            'author_id': message.author_id.display_name,
```

It's the same, but more readable. :wink:

Reply:
But we are not showing full display name on other parts...

Reply:
Please confirm this

Reply:
Yes, on other parts the module shows the same "name"...